### PR TITLE
Doc on how to use the state parameter

### PR DESCRIPTION
**Pull request by angelaLazar**
The goal of this new doc is to help explain to our users how to utilize the state parameter in a spec compliant way to help prevent XSRF attacks.

https://trello.com/c/3galf4YN/255-how-to-use-the-state-parameter-in-the-
spec-complaint-way

### CODE REVIEW

**Comment by auth0build** [comment]
Can one of the admins verify this patch?

File: articles/protocols/oauth-state.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+3. Send this value as the `state` parameter in the request.
+
+`tenant.auth0.com/authorize?...&state=xyzABC123`
+
+4. After the request is sent and when the user is redirected back by Auth0 to the client and the state value will be included. Note that depending on the type of connection used, this value might be in the body of the request or in the query string.
+
+`/login/callback?...&state=xyz123`
+
+5. Use this returned state value to compare to the previously stored value. If the values match then approve the request, else deny the request. Using this field for authentication of the request you can help prevent cross-site request forgery.
```
Comment (jaredhanson) [inline articles/protocols/oauth-state.md]: > Using this field for authentication of the request you can help prevent cross-site request forgery.

I would rewrite this to: "Using this field can help prevent cross-site request forgery".  The field isn't for authentication, and its potentially confusing to refer to that.

File: articles/protocols/oauth-state.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+description: Explains how to use the state parameter in authentication requests to help prevent XSRF attacks.
+---
+
+# Using the State Parameter
+
+The `state` parameter is one the supported Auth0 [Authentication Parameters](/libraries/lock/v10/sending-authentication-parameters). This page is to help you understand how to utilize this this parameter which is useful to help mitigate [XSRF attacks](https://en.wikipedia.org/wiki/Cross-site_request_forgery) and for providing any contextual information (such as a return url) that you might need after the authentication process is finished.
```
Comment (jaredhanson) [inline articles/protocols/oauth-state.md]: "one of"

File: updates/2016-08-16.yml [hunk 3]
```diff
@@ -1,3 +1,10 @@
+added:
```
Comment (mpaktiti) [inline updates/2016-08-16.yml:1]: Hey @angelaLazar this is an old file, can you move the new content to a new `updates/2016-10-17.yml` file? Thanks.

**Comment by mpaktiti** [comment]
Replaced by https://github.com/auth0/docs/pull/2561

**Pull request closed without merging**